design/src/rfcs/rfc0023_serialization_and_deserialization.md

```diff
@@ -0,0 +1,295 @@
+
+Currently, SDK has no data type that represents an entire response or request that can be mapped to `serde`'s data model; Thus, you must introduce a schema and implement logics that allows users to serialize/de-serialize their data.
+
+Although this RFC does not solve this issue, we believe that this RFC will help future contirbutor who wishes to implement serialization and de-serialization support for an entire response/request.
+
+
+## Sensitive Information
+If serialized data contains sensitive information, it will not be masked.  
+Users must be cautious to avoid compromisation
+
+## Compile Time
+We ran the following benchmark on C6a.2xlarge instance with 50gb of GP2 SSD.  
+The commit hash of the code is a8e2e19129aead4fbc8cf0e3d34df0188a62de9f.
+
+It clearly shows an increase in compile time.  
+Users are advised to consider the use of software such as [sccache](https://github.com/mozilla/sccache) or [mold](https://github.com/rui314/mold) to reduce the compile time.
+
+- `aws-sdk-dynamodb`
+ 
+| command                                                     | real time | user time | sys time  |
+| ----------------------------------------------------------- | --------- | --------- | --------- |
+| cargo build                                                 | 0m35.728s | 2m24.243s | 0m11.868s |
+| cargo build --release                                       | 0m52.040s | 5m0.841s  | 0m11.313s |
+| cargo build --features unstable-serde-serialize             | 0m38.079s | 2m26.082s | 0m11.631s |
+| cargo build --release --features unstable-serde-serialize   | 0m53.153s | 5m4.069s  | 0m11.577s |
+| cargo build --features unstable-serde-deserialize           | 0m45.689s | 2m34.000s | 0m11.978s |
+| cargo build --release --features unstable-serde-deserialize | 1m0.107s  | 5m10.231s | 0m11.699s |
+| cargo build --all-features                                  | 0m48.959s | 2m45.688s | 0m13.359s |
+| cargo build --release --all-features                        | 1m3.198s  | 5m26.076s | 0m12.311s |
+
+- `aws-sdk-ec2`
+ 
+| command                                                     | real time | user time  | sys time  |
+| ----------------------------------------------------------- | --------- | ---------- | --------- |
+| cargo build                                                 | 1m20.041s | 2m14.592s  | 0m6.611s  |
+| cargo build --release                                       | 2m29.480s | 9m19.530s  | 0m15.957s |
+| cargo build --features unstable-serde-serialize             | 2m0.555s  | 4m24.881s  | 0m16.131s |
+| cargo build --release --features unstable-serde-serialize   | 2m45.002s | 9m43.098s  | 0m16.886s |
+| cargo build --features unstable-serde-deserialize           | 3m10.857s | 5m34.246s  | 0m18.844s |
+| cargo build --release --features unstable-serde-deserialize | 3m47.531s | 10m52.017s | 0m18.404s |
+| cargo build --all-features                                  | 3m31.473s | 6m1.052s   | 0m19.681s |
```

Comment:
can we create DLL with rust?

Reply:
We'd reorganize all the SDK crates so that everything related to an operation could be activated/deactivated with a single feature. Users would decide which operations they wanted to use and then add the names of those operations as feature. We have a [separate RFC describing it in detail](https://github.com/awslabs/smithy-rs/blob/main/design/src/rfcs/rfc0026_client_crate_organization.md).

Reply:
Cool, thanks!